kube/src/api/dynamic.rs

```diff
@@ -56,125 +56,38 @@ impl GroupVersionKind {
         } else {
             format!("{}/{}", group, version)
         };
-        let plural = Some(ar.name.clone());
-        Self {
+        let plural_name = ar.name.clone();
+        ApiResource {
             group,
             version,
             kind,
             api_version,
-            plural,
+            plural_name,
         }
     }
 
-    /// Set the api group, version, and kind for a resource
-    pub fn gvk(group_: &str, version_: &str, kind_: &str) -> Result<Self> {
-        let version = version_.to_string();
-        let group = group_.to_string();
-        let kind = kind_.to_string();
-        let api_version = if group.is_empty() {
-            version.to_string()
-        } else {
-            format!("{}/{}", group, version)
+    /// Creates ApiResource from group, version and kind.
+    /// # Warning
+    /// This function has to **guess** resource plural name.
+    /// While it makes it best to guess correctly, sometimes it can
+    /// be wrong, and using returned ApiResource will lead to incorrect
```

Comment:
Hmm, not super happy about this yet. The plural builder before at least let people inject a correct value, but now it looks like it only works well with apidiscovery. We need to do something for kube-derive case which was using GVK with plural.

Maybe we should have a way to retrieve the ApiResource from a derived CustomResource? That way people wouldn't have to do the error prone setup of inserting a custom plural twice.

Reply:
> inject a correct value

It can be achieved by manually setting `plural_name` on created ApiResource. Do you think highlighting this in docs is enough? Alternatively, we can add a `plural: Option<String>` parameter to `from_gvk`.

> Maybe we should have a way to retrieve the ApiResource from a derived CustomResource

Very good point. CustomResource should have something like `api_resource() -> ApiResource`.

Reply:
I think what you've done for kube-derive makes it good enough to not worry about plural `from_gvk`.

I don't think it's worth keeping the old plural behaviour in the conversion if people can make a more accurate thing auto-generated.

One thing that comes up is to move the generated methods behind a trait, but I'll make a issue for that separately.

Reply:
Actually, if we are in general not encouraging `from_gvk` and not turning it into an `TryFrom` or `From` impl, then it probably makes sense to have a way to do it explicitly.

Maybe a `from_gvk_plural`, and leave `from_gvk` to infer?

Reply:
Blue-sky thinking.. another way would be to deprecate inferred `from_gvk`, and instead use the discovery machinery to look up the pluralization?

Reply:
> discovery helper to look up the details based on GVK

I think this [method](https://github.com/clux/kube-rs/blob/4f16ae057e788ca9e8d85ce199c895c0cd85d29a/kube/src/client/discovery.rs#L123) implements it. I also think deprecation has some sense, but I'll better leave this decision to you and other maintainers:)